Applications/Viqui/vqCore.cxx

```diff
@@ -3387,6 +3396,43 @@ void vqCore::generateReport(QString path, bool generateVideo)
   this->exportKml(path);
 }
 
+//-----------------------------------------------------------------------------
+void vqCore::exportIqr()
+{
+  this->updateIqrModel();
+
+  QString path =
+    vgFileDialog::getSaveFileName(
+      nullptr, "Save IQR model...", QString(), "IQR File (*.iqr);;");
+
+  if (path.isEmpty())
+    {
+    return;
+    }
+
+  QFile file(path);
+  if (!file.open(QIODevice::WriteOnly |
+                 QIODevice::Truncate))
+    {
+    qDebug() << file.errorString();
+    QMessageBox::warning(nullptr, QString(), "Unable to write IQR model file.");
+    return;
+    }
+
+  const auto& query = *this->SavedQueryPlan.constSimilarityQuery();
+
+  if (!query.IqrModel.empty())
```

Comment:
Please check this *before* asking the user for the file name so that we do not write an empty file. (And report that the IQR model is empty.)